op-node/rollup/engine/build_seal.go

```diff
@@ -0,0 +1,117 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+)
+
+// PayloadSealInvalidEvent identifies a permanent in-consensus problem with the payload sealing.
+type PayloadSealInvalidEvent struct {
+	Info eth.PayloadInfo
+	Err  error
+
+	IsLastInSpan bool
+	DerivedFrom  eth.L1BlockRef
+}
+
+func (ev PayloadSealInvalidEvent) String() string {
+	return "payload-seal-invalid"
+}
+
+// PayloadSealTemporaryErrorEvent identifies temporarily failed payload-sealing.
+// The user should re-attempt by starting a new build process.
+type PayloadSealTemporaryErrorEvent struct {
+	Info eth.PayloadInfo
+	Err  error
+
+	IsLastInSpan bool
+	DerivedFrom  eth.L1BlockRef
+}
+
+func (ev PayloadSealTemporaryErrorEvent) String() string {
+	return "payload-seal-temporary-error"
+}
+
+type BuildSealEvent struct {
+	Info         eth.PayloadInfo
+	BuildStarted time.Time
+	// if payload should be promoted to safe (must also be pending safe, see DerivedFrom)
+	IsLastInSpan bool
+	// payload is promoted to pending-safe if non-zero
+	DerivedFrom eth.L1BlockRef
+}
+
+func (ev BuildSealEvent) String() string {
+	return "build-seal"
+}
+
+func (eq *EngDeriver) onBuildSeal(ev BuildSealEvent) {
+	ctx, cancel := context.WithTimeout(eq.ctx, buildSealTimeout)
+	defer cancel()
+
+	sealingStart := time.Now()
+	envelope, err := eq.ec.engine.GetPayload(ctx, ev.Info)
+	if err != nil {
+		if x, ok := err.(eth.InputError); ok && x.Code == eth.UnknownPayload { //nolint:all
+			eq.log.Warn("Cannot seal block, payload ID is unknown",
+				"payloadID", ev.Info.ID, "payload_time", ev.Info.Timestamp,
+				"started_time", ev.BuildStarted)
+		}
+		// As verifier it is safe to ignore this event, attributes will be re-attempted,
+		// and any invalid-attributes error should be raised upon
+		// the start of block-building and/or later block insertion.
+		eq.emitter.Emit(PayloadSealTemporaryErrorEvent{
```

Comment:
It's "temporary" in the sense that the exact same payload-attributes could still be valid. I.e. there's not necessarily an in-protocol consensus issue with the input. The sequencer will give up, and restart building. Since if it did hit a timeout, it will likely also have expired the block-building job, so it will not be retrievable again.
I'll clarify the comments around this code and in the sequencer.